### PR TITLE
feat: better signature checking

### PR DESCRIPTION
**Pull request by m30m**
- Better error message when we know which signature is not valid
- Check that relayer pubkey is actually in the tx as a signer
- Better error if no opportunity was found

### CODE REVIEW

File: auction-server/src/auction.rs [hunk 1]
```diff
@@ -1249,21 +1265,20 @@ async fn verify_signatures_svm(
                     ),
                 })
                 .await;
-            opportunities.into_iter().any(|opportunity| {
-                let mut missing_signers = opportunity.get_missing_signers();
-                missing_signers.push(*relayer_pubkey);
-                all_signature_exists_svm(&message_bytes, accounts, &signatures, &missing_signers)
-            })
+            let opportunity = opportunities
+                .first()
```
Comment (anihamde) [inline auction-server/src/auction.rs:1269]: i assume you're trying to check the most recent opportunity added? just make sure that first() returns the most recent and not the oldest

File: auction-server/src/auction.rs [hunk 2]
```diff
@@ -1249,21 +1265,20 @@ async fn verify_signatures_svm(
                     ),
                 })
                 .await;
-            opportunities.into_iter().any(|opportunity| {
-                let mut missing_signers = opportunity.get_missing_signers();
-                missing_signers.push(*relayer_pubkey);
-                all_signature_exists_svm(&message_bytes, accounts, &signatures, &missing_signers)
-            })
+            let opportunity = opportunities
```
Comment (anihamde) [inline auction-server/src/auction.rs:1268]: just a note: why do we check against existing opportunities? there are some tradeoffs to just simulating the tx and not checking that an opp exists for this PK, but that's how we handle on evm side, i don't recall why we decided on checking against existing opps for svm

**Review by anihamde — APPROVED** [review]

File: auction-server/src/auction.rs (hunk 2 above)
Comment (m30m) [reply auction-server/src/auction.rs:1268]: main reason here is to ignore the other necessary signature from wallet which comes at the end of the flow

File: auction-server/src/auction.rs (hunk 1 above)
Comment (m30m) [reply auction-server/src/auction.rs:1269]: not really. There should be only one opportunity but the APIs return `Vec`.
@danimhr please correct me if I'm wrong

Comment (guibescos) [reply auction-server/src/auction.rs:1269]: mmm I'm not sure https://github.com/pyth-network/per/blob/main/auction-server/src/opportunity/repository/add_opportunity.rs#L41

Comment (m30m) [reply auction-server/src/auction.rs:1271]: I added some comments here

**Pull request merged**